roles/ceph-osd/tasks/pre_requisite.yml

```diff
@@ -58,3 +58,23 @@
   when:
     - cephx
     - item.copy_key|bool
+
+- set_fact:
+     osd_uuids: |
+      {%- set o=[] %}
+      {%- for device in  devices %}
+        {%- if o.append( ( inventory_hostname + device + ansible_date_time.iso8601_micro + "osd_uuids") | to_uuid ) %}
```

Comment:
the benefit of registering an UUID ourselve is to slove the follow FAQ

> FAQ
I want to have OSD numbers seriallized between hosts, so the first OSD node has osd 1,2,3 and the second has osd 4,5,6 etc. How can I do this? Simply add serial: 1 after the osd section - hosts: osds in your site.yml file.

when we have Hundreds osd servers and want to keep  OSD numbers seriallized between hosts. register osd id first and then prepare and actiavte osds  could be much more faster than the follow 

```
- hosts: osds
  become: True
  serial: 1
  roles:
    - ceph-osd
```